lib/template-loader.php

```diff
@@ -370,15 +369,15 @@ function gutenberg_strip_php_suffix( $template_file ) {
  * @return array Filtered editor settings.
  */
 function gutenberg_template_loader_filter_block_editor_settings( $settings ) {
-	global $post, $_wp_current_template_id;
+	global $post;
 
 	if ( ! $post || ! post_type_exists( 'wp_template' ) || ! post_type_exists( 'wp_template_part' ) ) {
 		return $settings;
 	}
 
 	// Create template part auto-drafts for the edited post.
-	$post = isset( $_wp_current_template_id )
-		? get_post( $_wp_current_template_id ) // It's a template.
+	$post = isset( $settings['templateId'] )
```

Comment:
I think this is only set in `edit-site`. This would break the post editor with template parts then.

Reply:
Revisiting this, I'm not sure we actually need the ternary checking for `$_wp_current_template_id` at all.

That global is (only) [set](https://github.com/WordPress/gutenberg/blob/9c256e3ae8bb715a9a863e7081df479acde71666/lib/template-loader.php#L99) by [`gutenberg_override_query_template()`](https://github.com/WordPress/gutenberg/blob/9c256e3ae8bb715a9a863e7081df479acde71666/lib/template-loader.php#L93), which ([via](https://github.com/WordPress/gutenberg/blob/9c256e3ae8bb715a9a863e7081df479acde71666/lib/template-loader.php#L96) [`gutenberg_find_template_post_and_parts()`](https://github.com/WordPress/gutenberg/blob/9c256e3ae8bb715a9a863e7081df479acde71666/lib/template-loader.php#L216)) also [calls](https://github.com/WordPress/gutenberg/blob/9c256e3ae8bb715a9a863e7081df479acde71666/lib/template-loader.php#L340) `create_auto_draft_for_template_part_block()`. So the template case should be covered, leaving only the post editor case.

Reply:
But that sounds wrong, since this is clearly targeted to the editor case, and `gutenberg_override_query_template()` is only called for front-end template resolution. That might be an oversight of my part going back to #21959 :confused:

Reply:
However, [`gutenberg_find_template_post_and_parts`](https://github.com/WordPress/gutenberg/blob/9c256e3ae8bb715a9a863e7081df479acde71666/lib/edit-site-page.php#L156) is called for all template types in `lib/edit-site-page.php`, so template part auto-drafts are indeed auto-generated for the site editor.

Reply:
How would this make sense in the post editor anyways? i.e. how would there be a template set for an individual post? And if there is a template set, why do we care about the template parts associated with the template (at this point in time)? Wouldn't we only care about template parts inside the post_content of the post?

Reply:
> Wouldn't we only care about template parts inside the post_content of the post?

Exactly, yeah :thinking:

Reply:
Pushed d6dc5a3df1 to reflect these musings in code.

Reply:
I think this is leftover from when you could switch to a template view in the post editor.